programs/gzip.c

```diff
@@ -258,9 +261,11 @@ do_decompress(struct libdeflate_decompressor *decompressor,
 			goto out;
 		}
 
-		ret = full_write(out, uncompressed_data, actual_out_nbytes);
-		if (ret != 0)
-			goto out;
+		if (!options->test) {
+			ret = full_write(out, uncompressed_data, actual_out_nbytes);
+			if (ret != 0)
+				goto out;
+		} /* "else" (test): as do_decompress() finished successfully it means file was decompressed properly */
```

Comment:
This comment is wrong since `do_decompress()` hasn't finished successfully yet.

I don't think it adds anything; it would be better to just remove this comment.